Feat/migrate tag and selectlist css vars

martimalek opened the pull request:
**What:**

Migrate new variants of `Tag` and `SelectList` components to use css vars for colors.

​
**Why:**

The `Tag` and `SelectList` components have recently received some updates in `main`, since in `main` the css variables are not yet available these updates were done using the "old" colors constants. We need to migrate to the new css vars here in `next`.

​
**How:**

I've migrated all color values in `Tag` and `SelectList` so they are using `getSemanticValue`. Please review the semantic values that I've chosen and let me know if you think that there are better alternatives, as far as I understand, we don't have a way to perfectly map the old semantic colors to the new ones, I've tried to keep them semantic.

​
**Media:**

`Tag`

<img width="585" alt="image" src="https://github.com/freenowtech/wave/assets/46452321/a92c36f6-c55b-42b7-bf66-898e5e90769c">

`SelectList` (multi)

<img width="565" alt="image" src="https://github.com/freenowtech/wave/assets/46452321/b2dd8452-ff0e-4e11-b500-cd433d3f515d">


**Checklist:**

-   [x] Ready to be merged

Closes #373

nlopin commented:
Those two components have semantic colors in the next branch (see [Tag](https://github.com/freenowtech/wave/blob/258d8ed5415c943bd94eace78615b92b660a101e/src/components/Tag/Tag.tsx#L43C1-L43C1)). In the main, error states were added.

Have you used the same semantic values as in the next branch? That way, we can avoid conflicts

martimalek commented:
The force pushes that you see are because I corrected the git history of #380, when merging `main` into it I renamed the merge commit by mistake, which made me lose the commits history between `main` and `next`. Now that it is corrected there I'm just updating it here.

martimalek commented:
> Those two components have semantic colors in the next branch (see [Tag](https://github.com/freenowtech/wave/blob/258d8ed5415c943bd94eace78615b92b660a101e/src/components/Tag/Tag.tsx#L43C1-L43C1)). In the main, error states were added.
> 
> Have you used the same semantic values as in the next branch? That way, we can avoid conflicts

Yes, I've used the same semantic colors that were already in `next`, I've only added the ones that were missing for the new variants
